syft/frameworks/types.py

```diff
@@ -0,0 +1,38 @@
+# from __future__ import absolute_import
+from typing import Union
+
+from syft import dependency_check
+
+framework_tensors = []
+framework_shapes = []
+hooks = []
+
+if dependency_check.tensorflow_available:
+    import tensorflow as tf
+    from tensorflow.python.framework.ops import EagerTensor
+
+    framework_tensors.append(EagerTensor)
+    framework_shapes.append(tf.TensorShape)
+
+if dependency_check.torch_available:
+    import torch
+
+    framework_tensors.append(torch.Tensor)
+    framework_shapes.append(torch.Size)
+
+FrameworkTensorType = None
+for tensor_type in framework_tensors:
```

Comment:
Is there a better way to do this by operating on the list?

Reply:
I couldn't find any -- the `Union` object is pretty rigid

Reply:
would it make sense to define it above as

and extend within `if dependency_check.torch_available:`
```py
FrameworkTensorType = Union[FrameworkTensorType, torch.Tensor]
```

the same for sizes

Reply:
>would it make sense to define it above as

I would end up repeating that code for each framework -- this way you just create an if block for each new framework that adds its own FrameworkTensorType(s) and FrameworkShapeType and the rest of the code doesn't have to change.

Reply:
@jvmancuso we can do this as FrameworkTensorType = Union[*framework_tensors]

Reply:
That's what I tried originally, but that assignment will fail with a SyntaxError. I suspected it has to do with the `*` operator only being interpretable at runtime, although I didn't actually check to verify that.

Reply:
Funny enough it work with 
```python
Union[(*framework_tensors)]
```

Reply:
![image](https://user-images.githubusercontent.com/775466/63290723-5c548e00-c2c2-11e9-85ef-db983c9d4722.png)

Reply:
or 
```python
Union[tuple(t)]
```
![image](https://user-images.githubusercontent.com/775466/63290873-a89fce00-c2c2-11e9-8ee6-5996d8e05173.png)

Reply:
nice one! will fix that up now